src/test/java/ca/on/oicr/gsi/dimsum/service/filtering/RunFilterTest.java

```diff
@@ -0,0 +1,40 @@
+package ca.on.oicr.gsi.dimsum.service.filtering;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import ca.on.oicr.gsi.dimsum.data.Run;
+
+public class RunFilterTest {
+
+
+  private static List<Run> runs =
+      Arrays.asList(makeRun("RUN1"),
+          makeRun("RUN2"), makeRun("RUN3"), makeRun("RUN4"),
+          makeRun("RUN5"));
+
+  @Test
+  public void testFilterName() {
+    RunFilter filter = new RunFilter(RunFilterKey.NAME, "RUN1");
+    testFilter(filter, Arrays.asList("RUN1"));
+  }
+
+  private static void testFilter(RunFilter filter, List<String> expectedNames) {
+    List<Run> filtered = runs.stream().filter(filter.predicate()).toList();
```

Comment:
Is that the syntax used for filtering a List of Run throughout the project? Could RunFilter have a method for applying itself to a List<Run>?

Reply:
Applying a single filter like this is only useful for testing. Filters are normally combined and applied in `CaseService`